src/helpers.ts

```diff
@@ -78,45 +94,50 @@ export const units = {
     },
 };
 
-const logUnitMap = {
-    0: units.B,
-    1: units.KiB,
-    2: units.MiB,
-    3: units.GiB,
-    4: units.TiB,
-    5: units.PiB,
-    6: units.EiB,
-};
+const logUnitMap = [
+    units.B,
+    units.KiB,
+    units.MiB,
+    units.GiB,
+    units.TiB,
+    units.PiB,
+    units.EiB,
+];
 
-function getPowerOf1024(exponent) {
+function getPowerOf1024(exponent: number): number {
     return exponent === 0 ? 1 : Math.pow(1024, exponent);
 }
 
-function getLogarithmOfBase1024(value) {
+function getLogarithmOfBase1024(value: number): number {
     return value > 0 ? (Math.floor(Math.log(value) / Math.log(1024))) : 0;
 }
 
-export function getBestUnit(input, inputUnit) {
+export function getBestUnit(input: unknown, inputUnit: string | Unit): Unit {
     return logUnitMap[getLogarithmOfBase1024(convertToUnitVerbose(input, inputUnit, units.B).value)];
 }
 
-export function convertToUnit(input, inputUnit, outputUnit) {
+export function convertToUnit(input: unknown, inputUnit: string | Unit, outputUnit: string | Unit): number {
     return convertToUnitVerbose(input, inputUnit, outputUnit).value;
 }
 
-function convertToUnitVerbose(input, inputUnit, outputUnit) {
-    const result = {
+interface Result {
+    value: number;
+    unit: string;
+}
+
+function convertToUnitVerbose(input: unknown, inputUnit: string | Unit, outputUnit: string | Unit): Result {
+    const result: Result = {
         value: 0,
         unit: units.B.name,
     };
 
-    input = Number(input);
-    if (isNaN(input)) {
+    const _input = Number(input);
```

Comment:
this is a nitpick from my side but I don't like `_input`, `parsed_input` or something else

Reply:
Sure.